pkg/cache/cache.go

```diff
@@ -0,0 +1,30 @@
+package cache
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrStaleCache = errors.New("item's ttl has expired")
```

Comment:
Probably `ErrNotFound` is enough - from outside of cache, the item is then not found, as it's expired. `ErrStaleCache` is implementation-dependent - it would not happen if i.e. obsolete items would be cleared externally, like in worker, or by key-value store TTL.

Reply:
My original idea was to be able to detect is item missing from cache or is it stale, and then you could do different actions, but to be fair, we don't have an use case for it, so it could be removed and to stick only with `ErrNotFound`